### ISSUE TITLE
Issue while setting up HTTP Proxy

### ISSUE THREAD

**Issue by subhasisbanik** [raiden-network/microraiden#111]
I am trying to set up the HTTP Proxy for microraiden following the steps in :
https://github.com/raiden-network/microraiden/blob/master/microraiden/README.md
 But while running the below command,
python3 -m microraiden.examples.demo_proxy --private-key <private_key_file> start

with a valid ethereum private key file,
I am facing the bleow issue:
"Error while finding spec for 'microraiden.examples.demo_proxy' (<class 'ImportError'>: No module named 'eth_abi')".

**Comment by pcppcp** [comment]
Hi!
Do you have all dependencies installed? At least `ethereum-abi-utils` is missing. 
It's recommended to set up `virtualenv` and then install all requirements with `pip install -r requirements.txt`.

**Comment by subhasisbanik** [comment]
I am running the below steps:
cd microraiden
virtualenv -p python3 env
. env/bin/activate
pip install -e microraiden
cd microraiden
git clone https://github.com/pipermerriam/ethereum-abi-utils.git
cd ethereum-abi-utils/
python setup.py install
cd ..
python -m microraiden.examples.demo_proxy --private-key <private_key_file> start
The last line now throws error,
"CRITICAL:microraiden.click_helpers:Private key file /home/ubuntu/microraiden/privateKey.txt must be readable only by its owner."

**Comment by andrevmatos** [comment]
@subhasisbanik as the error suggests, your privatekey file must have 600 or 700 permission, for security reasons.

**Comment by pcppcp** [comment]
change your private key permissions:
`chmod 600 /home/ubuntu/microraiden/privateKey.txt`

**Comment by subhasisbanik** [comment]
Thanks. that works.
But after running,
python -m microraiden.examples.demo_proxy --private-key ../privateKey.txt start
Now I am getting new issue like,
"WARNING:microraiden.utils:Private key in raw format. Consider switching to JSON-encoded
Traceback (most recent call last):
  File "/usr/lib/python3.4/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.4/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/ubuntu/uraiden/microraiden/microraiden/microraiden/examples/demo_proxy/__main__.py", line 55, in <module>
    main()
  File "/usr/local/lib/python3.4/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.4/dist-packages/click/core.py", line 1063, in invoke
    Command.invoke(self, ctx)
  File "/usr/local/lib/python3.4/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.4/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/ubuntu/uraiden/microraiden/microraiden/microraiden/click_helpers.py", line 104, in main
    app = make_paywalled_proxy(private_key, state_file, web3=web3)
  File "/home/ubuntu/uraiden/microraiden/microraiden/microraiden/make_helpers.py", line 57, in make_paywalled_proxy
    channel_manager = make_channel_manager(private_key, state_filename, web3)
  File "/home/ubuntu/uraiden/microraiden/microraiden/microraiden/make_helpers.py", line 38, in make_channel_manager
    state_filename=state_filename
  File "/home/ubuntu/uraiden/microraiden/microraiden/microraiden/channel_manager.py", line 313, in __init__
    self.state = ChannelManagerState.load(state_filename)
  File "/home/ubuntu/uraiden/microraiden/microraiden/microraiden/channel_manager.py", line 270, in load
    json_state = json.loads(open(filename, 'r').read())
  File "/usr/lib/python3.4/json/__init__.py", line 318, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.4/json/decoder.py", line 343, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.4/json/decoder.py", line 361, in raw_decode
    raise ValueError(errmsg("Expecting value", s, err.value)) from None
ValueError: Expecting value: line 1 column 1 (char 0)"


My private key is,
0x24236464535(in this format)

**Comment by pcppcp** [comment]
try to remove files in `/home/ubuntu/.config/microraiden/`, this sounds like a problem with balance state.

**Comment by subhasisbanik** [comment]
Thanks.
Now I am getting logs as below:
INFO:filelock:Lock 139960717454192 acquired on /home/ubuntu/.config/microraiden/0xffa52825_0x8b0d76e0.json
DEBUG:channel_manager:setting up channel manager, receiver=0x8b0d76e01ecc08070b140bc431d4e7834e22a691 channel_contract=0xffa52825c7997dd2be80fb91080500a52abd6d5b
INFO:blockchain:starting blockchain polling (interval 2s)
DEBUG:blockchain:filtering for events u:0-3 c:0-2 @3
INFO:blockchain:initial sync finished

Is there any command line args to set the port for viewing the webpage?

**Comment by pcppcp** [comment]
Congrats, now you've got it working. 
One thing that strikes me is the block number - is this a custom chain?

You can set up the host and port as follows:
`python -m microraiden.examples.demo_proxy --private-key ../privateKey.txt start --host localhost --port 8081`

**Comment by subhasisbanik** [comment]
@pcppcp I have provided a generic genesis.json. I want to connect to the main chain to be able to do wallet operations. Does this mean otherwise?

Moreover, I am getting log with your command like,

INFO:filelock:Lock 140391471725480 acquired on /home/ubuntu/.config/microraiden/0xffa52825_0x8b0d76e0.json
DEBUG:channel_manager:setting up channel manager, receiver=0x8b0d76e01ecc08070b140bc431d4e7834e22a691 channel_contract=0xffa52825c7997dd2be80fb91080500a52abd6d5b
INFO:blockchain:starting blockchain polling (interval 2s)

Is this correct?

**Comment by pcppcp** [comment]
Yes this appears to be correct. But you must deploy the contracts first if you are using custom chain - now you are using channel manager contract that's deployed on Kovan.

**Comment by subhasisbanik** [comment]
I want to use the public chain to be able to deal with real tokens. Should I go ahead with my approach? @pcppcp

**Comment by pcppcp** [comment]
Yes you can run it also on a public chain. The default configuration is for Kovan chain, but there are contracts deployed on Ropsten as well.

You can either run parity node on your machine, or use Infura infrastructure:
`python3 -m microraiden.examples.demo_proxy --rpc-provider https://kovan.infura.io/ --private-key /path/to/key start`

**Comment by subhasisbanik** [comment]
Now I am able to connect to microraiden from parity. Even I am able to run the doggo.jpg app.
I have also funded my account with certain amount of tokens with Ether.
Now it says there is no channel and I am prompted to depot 2 tokens to create a channel and view the document.But whenever I am clicking on deposit, I am getting a prompt from Metamask and after clicking on Submit, I do not get a response even after waiting for long.
Moreover, I tried to check the console and found a transaction hash and searched for the same on the explorer to know that the transaction is successful and got X no. of confirmations. 
I even tried to check the network calls but didn't find any.
Below are the addresses present in the microraiden:
CHANNEL_MANAGER_ADDRESS = '0xffa52825c7997dd2be80fb91080500a52abd6d5b'
TOKEN_ADDRESS = '0x04c7f744a0c751d89e99ae79a39060ec6f3c4397'
Is there anything that I am doing wrong in this?
